src/Uno.UI/UI/Xaml/Media/LinearGradiientBrush.faux.cs

```diff
@@ -0,0 +1,120 @@
+﻿#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Linq;
+using Uno.UI.Xaml.Media;
+using Windows.Foundation.Collections;
+using Windows.UI;
+
+namespace Microsoft.UI.Xaml.Media;
+
+public partial class LinearGradientBrush
+{
+	private Color? _majorStopColor;
+	private SolidColorBrush? _fauxOverlayBrush;
+
+	/// <summary>
+	/// Gets or sets a flag indicating whether this brush should use
+	/// major gradient solid brush color in case the target platform
+	/// cannot apply the linear gradient to border correctly.
+	/// </summary>
+	public bool SupportsFauxBorder
+	{
+		get => (bool)GetValue(SupportsFauxBorderProperty);
+		set => SetValue(SupportsFauxBorderProperty, value);
+	}
+
+	/// <summary>
+	/// Identifies the SupportsFauxBorder dependency property.
+	/// </summary>
+	public static DependencyProperty SupportsFauxBorderProperty { get; } =
+		DependencyProperty.Register(
+			nameof(SupportsFauxBorder),
+			typeof(bool),
+			typeof(LinearGradientBrush),
+			new FrameworkPropertyMetadata(false));
+
+	internal Color? MajorStopColorWithOpacity => _majorStopColor ??= GetMajorStop()?.Color.WithOpacity(Opacity);
+
+	internal SolidColorBrush? FauxOverlayBrush => _fauxOverlayBrush ??= CreateFauxOverlayBrush();
```

Comment:
The gradient stops may change after we have cached this. I think we should invalidate cache when necessary (if there is a change in gradient stops, or when `SupportsFauxBorder` changes.

Reply:
This is not a problem, as we currently only use the faux overlay for the several built-in brushes which are just static, so we can just cache the fallback once for each of the brushes and keep it intact

Reply:
Yup should be good as it's internal. I was only curious when the API was public